worlds/jakanddaxter/__init__.py

```diff
@@ -0,0 +1,287 @@
+            count = 1
+
+        # Make N Precursor Orb bundles, where N is 2000 / bundle size.
+        elif item in range(jak1_id + Orbs.orb_offset, jak1_max):
+            if self.total_trade_orbs == 0:
+                classification = ItemClass.filler  # If you don't need orbs to do trades, they are useless.
+            else:
+                classification = ItemClass.progression_skip_balancing
+            count = 2000 // self.orb_bundle_size if self.orb_bundle_size > 0 else 0  # Don't divide by zero!
+
+        # Under normal circumstances, we will create 0 filler items.
+        # We will manually create filler items as needed.
+        elif item == jak1_max:
+            classification = ItemClass.filler
+            count = 0
+
+        # If we try to make items with ID's higher than we've defined, something has gone wrong.
+        else:
+            raise KeyError(f"Tried to fill item pool with unknown ID {item}.")
+
+        return count, classification
+
+    def create_items(self) -> None:
+        for item_name in self.item_name_to_id:
+            item_id = self.item_name_to_id[item_name]
+
+            # Handle Move Randomizer option.
+            # If it is OFF, put all moves in your starting inventory instead of the item pool,
+            # then fill the item pool with a corresponding amount of filler items.
+            if item_name in self.item_name_groups["Moves"] and not self.options.enable_move_randomizer:
+                self.multiworld.push_precollected(self.create_item(item_name))
+                self.multiworld.itempool.append(self.create_filler())
+                continue
+
+            # Handle Orbsanity option.
+            # If it is OFF, don't add any orb bundles to the item pool, period.
+            # If it is ON, don't add any orb bundles that don't match the chosen option.
+            if (item_name in self.item_name_groups["Precursor Orbs"]
+                and (self.options.enable_orbsanity == EnableOrbsanity.option_off
+                     or item_name != self.orb_bundle_item_name)):
+                continue
+
+            # In every other scenario, do this.
+            count, classification = self.item_type_helper(item_id)
+            self.multiworld.itempool += [JakAndDaxterItem(item_name, classification, item_id, self.player)
+                                         for _ in range(count)]
+
+    def create_item(self, name: str) -> Item:
+        item_id = self.item_name_to_id[name]
+        _, classification = self.item_type_helper(item_id)
+        return JakAndDaxterItem(name, classification, item_id, self.player)
+
+    def get_filler_item_name(self) -> str:
+        return "Green Eco Pill"
+
+    def collect(self, state: CollectionState, item: Item) -> bool:
+        change = super().collect(state, item)
+        if change:
+            # No matter the option, no matter the item, set the caches to stale.
+            state.prog_items[self.player]["Reachable Orbs Fresh"] = False
```

Comment:
Orb items do not immediately give you access to more reachable orbs (locations), correct?
If so, it is probably worth it to exclude orb items here, since receiving an orb should not change how many orb locations you have available. Unless I'm forgetting something that costs orbs that gives you access to more orb locations.

Reply:
Right, there's no _direct_ correlation where more orb locations are automatically unlocked by you receiving orb items. Like, Precursor Basin is not locked until you have 50 orbs, and no level or region _that contains orb locations_ is locked behind the mere possession of orbs.

Now, the _indirect_ correlation is that more orb locations may be unlocked if a trader has a key item and you just obtained enough orb items to trade with them. Then that key item unlocks a level or region with more orb locations.

Reply:
Tried that suggestion out on the big 50-person nightmare seed, but no difference it seems.

Reply:
If the key item you get from a trader unlocks more locations, then collecting that key item is what matters, not the orb itself. The orb itself does not give you access to more locations.

The 50-person nightmare seed is swap happening a lot, so this wouldn't speed that up at all

Reply:
Ok, good on both fronts.